docs: add installation guide (#818)
asyncapi/generator: docs/versioning.md

---
title: "Generator Version vs Template Version"
weight: 60

> ---

Author: This doc is about **the generator version vs the template version** but I found most of what I explained is about the generator version, not the template version. This is one thing I think can be improved in this doc. Maybe you can give me some pointers.

> The generator tool generates whatever you want, as long as it can be defined in a template, based on the [AsyncAPI file](asyncapi-file.md). On the other hand, a **template** is a file or group of files that specify the kind of output you expect from using the generators features. For example, you may want to use the [nodejs template](https://github.com/asyncapi/nodejs-template) to generate boilerplate code for your message based APIs.

Reviewer: ```suggestion
The generator tool generates whatever you want, as long as it can be defined in a template, based on the [AsyncAPI file](asyncapi-file.md). On the other hand, a **template** is a file or group of files that specify the kind of output you expect from using the generator's features. For example, you may want to use the [nodejs template](https://github.com/asyncapi/nodejs-template) to generate boilerplate code for your message-based APIs.
```

> The generator tool generates whatever you want, as long as it can be defined in a template, based on the [AsyncAPI file](asyncapi-file.md). On the other hand, a **template** is a file or group of files that specify the kind of output you expect from using the generator's features. For example, you may want to use the [nodejs template](https://github.com/asyncapi/nodejs-template) to generate boilerplate code for your message-based APIs.

Reviewer 2: ```suggestion
The generator tool generates whatever you want, as long as it can be defined in a template based on the [AsyncAPI file](asyncapi-file.md). On the other hand, a **template** is a file or group of files that specify the kind of output you expect from using the generator's features. For example, you may want to use the [NodeJS template](https://github.com/asyncapi/nodejs-template) to generate boilerplate code for your message-based APIs.
```

`defined in a template based` - maybe link to template too?

> Templates are dependent on the generators' features. For example, the template you want to use may be compatible with the latest version of the generator but incompatible with the previous versions. Check the configuration file or ReadMe of the template to see the version of the Generator it supports. The generator has a `isTemplateCompatible` function that checks if the template is compatible with the version of generator you want to use. If the template isn't compatible, you will see an error output similar to the following on your terminal:

Reviewer: ```suggestion
Templates are dependent on the generators' features. For example, the template you want to use may be compatible with the latest generator version but incompatible with the previous versions. Check the configuration file or ReadME of the template to see the version of the generator it supports. The generator has an `isTemplateCompatible` function that checks if the template is compatible with the version of the generator you want to use. If the template isn't compatible, you will see a terminal error output similar to the following:
```

```
Something went wrong:
Error: This template is not compatible with the current version of the generator (${generatorVersion}). This template is compatible with the following version range: ${generator}.`)
```

> > Use the following command to check the version of the AsyncAPI Generator CLI you have installed;  `ag --version`

Reviewer 2: ```suggestion
> Use the following command to check the version of the AsyncAPI CLI you have installed;  `asyncapi --version`
```

> It is better to lock a specific version of the template and the Generator if you plan to use the AsyncAPI CLI and a particular template in production. The differences between using the version of the AsyncAPI CLI you have installed and locking a certain version on production are demonstrated in the following code snippets.

Reviewer: ```suggestion
It is better to lock a specific version of the template and the generator if you plan to use the AsyncAPI CLI and a particular template in production. The differences between using the version of the AsyncAPI CLI you have installed and locking a certain version on production are demonstrated in the following code snippets.
```

Consistency is KEY 🔑 . You haven't been capitalizing `generator` mentions so far, so let's make them all match.

Reviewer: @Florence-Njeri wait, is this commit added? it looks like it's pending 🤔

> It is better to lock a specific version of the template and the Generator if you plan to use the Generator CLI and a particular template in production. The differences between using the version of the Generator CLI you have installed and locking a certain version on production are demonstrated in the following code snippets.

Reviewer 2: ```suggestion
It is better to lock a specific version of the template and the Generator if you plan to use the AsyncAPI CLI and a particular template in production. The differences between using the version of the AsyncAPI CLI you have installed and locking a certain version on production are demonstrated in the following code snippets.
```

> It is better to lock to a specific version of the template and the Generator if you plan to use the Generator CLI and a particular template in production. The differences between using the version of the Generator CLI you have installed and locking a certain version on production are demonstrated in the following code snippets.

Reviewer: ```suggestion
It is better to lock a specific version of the template and the Generator if you plan to use the Generator CLI and a particular template in production. The differences between using the version of the Generator CLI you have installed and locking a certain version on production are demonstrated in the following code snippets.
```

> Generate HTML with the latest AsyncAPI Generator CLI using the html-template.

Reviewer 2: ```suggestion
Generate HTML with the latest AsyncAPI CLI using the html-template.
```

> To generate HTML with latest AsyncAPI Generator CLI using the html-template.

Reviewer: ```suggestion
Generate HTML with the latest AsyncAPI Generator CLI using the html-template.
```

```
npm install -g @asyncapi/generator

> ag asyncapi.yaml @asyncapi/html-template -o ./docs

Reviewer 2: ```suggestion
asyncapi generate fromTemplate asyncapi.yaml @asyncapi/html-template -o ./docs
```

```

> To generate HTML using a partcular version of the AsyncAPI Generator CLI using the html-template.

Reviewer: ```suggestion
Generate HTML using a particular version of the AsyncAPI Generator CLI using the html-template.
```

```

> npm install -g @asyncapi/generator@0.50.0

Reviewer 2: ```suggestion
npm install -g @asyncapi/cli@0.20.0
```

> ag asyncapi.yaml @asyncapi/html-template@0.7.0 -o ./docs

Reviewer 2: ```suggestion
asyncapi generate fromTemplate asyncapi.yaml @asyncapi/html-template@0.7.0 -o ./docs
```
